pkg/continuoustest/client.go

```diff
@@ -153,6 +154,8 @@ func (c *Client) QueryRange(ctx context.Context, query string, start, end time.T
 	ctx, cancel := context.WithTimeout(ctx, c.cfg.ReadTimeout)
 	defer cancel()
 
+	ctx = querierapi.ContextWithReadConsistency(ctx, querierapi.ReadConsistencyStrong)
```

Comment:
The ctx is passed to the read client, which is just an HTTP client. The context will not be propagated, so this won't work. We should inject the HTTP Header instead (`X-Read-Consistency: strong`). I think we can easily do it configuring a Roundtripper on the HTTP client passed to the Prometheus client in `NewClient()`.

Reply:
I believe we handle that (extracting the consistency level into the header) in the `clientRoundTripper` below, don't we?

Reply:
You're right @narqo and updated unit tests also show that.

Reply:
> I believe we handle that (extracting the consistency level from context into the header) in the clientRoundTripper chunk below, don't we?

Sorry. I totally missed it.